app/models/hmpps_api/offender.rb

```diff
@@ -19,34 +17,26 @@ def awaiting_allocation_for
 
     delegate :code, :label, :active_since, to: :@category, prefix: :category, allow_nil: true
 
-    attr_accessor :prison_arrival_date, :sentence
-
-    attr_reader :first_name, :last_name, :prison_id, :offender_no, :cell_location, :complexity_level, :date_of_birth
-
-    # This list must only contain fields that are both supplied by
-    # https://api-dev.prison.service.justice.gov.uk/swagger-ui.html#//prisoners/getPrisonersOffenderNo
-    # and also by
-    # https://api-dev.prison.service.justice.gov.uk/swagger-ui.html#//locations/getOffendersAtLocationDescription
-    def initialize(api_payload, search_payload, category:, latest_temp_movement:, complexity_level:, booking_id:, prison_id:)
-      # It is expected that this method will be called by the subclass which
-      # will have been given a payload at the class level, and will call this
-      # method from it's own internal from_json
-      @first_name = api_payload.fetch('firstName')
-      @last_name = api_payload.fetch('lastName')
-      @offender_no = api_payload.fetch('offenderNo')
-      @convicted_status = api_payload['convictedStatus']
-      @date_of_birth = api_payload.fetch('dateOfBirth').to_date
+    attr_accessor :prison_arrival_date
+
+    attr_reader :first_name, :last_name, :prison_id, :offender_no, :cell_location, :complexity_level,
+                :date_of_birth, :sentence, :main_offence
+
+    def initialize(prison_api_payload:, search_api_payload:, category:, latest_temp_movement:, complexity_level:)
+      @first_name = search_api_payload.fetch('firstName')
+      @last_name = search_api_payload.fetch('lastName')
+      @offender_no = search_api_payload.fetch('prisonerNumber')
+      @convicted_status = prison_api_payload['convictedStatus']
```

Comment:
Just out of curiosity, why in some instances have we not used `.fetch`

Reply:
@Katy600 I'm not entirely sure the reason for it – but the `[]` accessor is useful when you want to gracefully return `nil` if that key doesn't exist. I guess there's potential for the `convictedStatus` field to be missing from the Prison API response... 🤷🏻

Watch this space, though, because MO-852 removes the `convictedStatus` field so that we're no longer dependent on `prison_api_payload` – so we can make one less API request 🙂